haskell/toolchain.bzl

```diff
@@ -111,9 +111,12 @@ def _haskell_toolchain_impl(ctx):
     ghc_binaries = {}
     for tool in _GHC_BINARIES:
         for file in ctx.files.tools:
-            if tool == paths.split_extension(file.basename)[0]:
```

Comment:
Shouldn't we try the one with the version set first to ensure that we pick the most specific one in case of conflict?

Reply:
Your call. In practice they are exactly the same.

Reply:
I'd invert them just in case they happen not to be the same then

Reply:
@regnat I realized the order of the checks doesn't matter. What matters is the order in which the `ctx.files.tools` are read.